thirdparty/kyaml/runfn/runfn.go

```diff
@@ -455,6 +367,28 @@ func getUIDGID(asCurrentUser bool, currentUser currentUserFunc) (string, error)
 	return fmt.Sprintf("%s:%s", u.Uid, u.Gid), nil
 }
 
+// getFunctionConfig will return the config that used in 'functionConfig' field in
+// function input. If r.FnConfigPath is provided, it will read and return it. Otherwise
+// return the 'config' generated from command line arguments.
+func (r *RunFns) getFunctionConfig(config *yaml.RNode) (*yaml.RNode, error) {
+	if r.FnConfigPath == "" {
+		return config, nil
+	}
+	f, err := os.Open(r.FnConfigPath)
+	if err != nil {
+		return nil, err
+	}
+	reader := kio.ByteReader{Reader: f}
+	nodes, err := reader.Read()
+	if err != nil {
+		return nil, err
+	}
+	if len(nodes) > 1 {
+		return nil, fmt.Errorf("more than 1 config found in %s", r.FnConfigPath)
+	}
+	return nodes[0], nil
+}
+
 // ffp provides function filters
 func (r *RunFns) ffp(spec runtimeutil.FunctionSpec, api *yaml.RNode, currentUser currentUserFunc) (kio.Filter, error) {
```

Comment:
Can we rename this function to `defaultFnProvider`

Reply:
Renamed to `defaultFnFilterProvider`